keyboards/sofle/keymaps/noqmk/keymap.c

```diff
@@ -0,0 +1,686 @@
+                                                     }};
+
+    /* animation */
+    void animate_luna(void) {
+        /* jump */
+        if (isJumping || !showedJump) {
+            /* clear */
+            oled_set_cursor(LUNA_X, LUNA_Y + 2);
+            oled_write("     ", false);
+
+            oled_set_cursor(LUNA_X, LUNA_Y - 1);
+
+            showedJump = true;
+        } else {
+            /* clear */
+            oled_set_cursor(LUNA_X, LUNA_Y - 1);
+            oled_write("     ", false);
+
+            oled_set_cursor(LUNA_X, LUNA_Y);
+        }
+
+        /* switch frame */
+        current_frame = (current_frame + 1) % 2;
+
+        /* current status */
+        if (led_usb_state.caps_lock) {
+            oled_write_raw_P(bark[abs(1 - current_frame)], ANIM_SIZE);
+
+        } else if (isSneaking) {
+            oled_write_raw_P(sneak[abs(1 - current_frame)], ANIM_SIZE);
+
+        } else if (current_wpm <= MIN_WALK_SPEED) {
+            oled_write_raw_P(sit[abs(1 - current_frame)], ANIM_SIZE);
+
+        } else if (current_wpm <= MIN_RUN_SPEED) {
+            oled_write_raw_P(walk[abs(1 - current_frame)], ANIM_SIZE);
+
+        } else {
+            oled_write_raw_P(run[abs(1 - current_frame)], ANIM_SIZE);
+        }
+    }
+
+    /* animation timer, stops the animation logic when the oled is turned off  */
+    if (timer_elapsed32(anim_timer) > ANIM_FRAME_DURATION && animate == true) {
+        anim_timer = timer_read32();
+        animate_luna();
+    }
+
+    /* this fixes the screen on and off bug by disabling the animation logic when the oled is off */
+    if (current_wpm > 0) {
+        oled_on();
+        anim_sleep = timer_read32();
+        animate = true;
+    } else if (timer_elapsed32(anim_sleep) > OLED_TIMEOUT) {
+        oled_off();
+        animate = false;
+    }
+}
+
+/* KEYBOARD PET END */
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    if (is_keyboard_master()) {
+        return OLED_ROTATION_270;
+    }
+    return rotation;
+}
+
+static void print_logo_narrow(void) {
+    render_logo();
+}
+
+static void print_status_narrow(void) {
+    /* Print current mode */
+    oled_set_cursor(0, 0);
+    if (show_lock == true) {
+        oled_write_raw_P(locked_logo, sizeof(locked_logo));
+    } else if (keymap_config.swap_lctl_lgui) {
+        oled_write_raw_P(mac_logo, sizeof(mac_logo));
+    } else {
+        oled_write_raw_P(linux_logo, sizeof(linux_logo));
+    }
+
+    oled_set_cursor(0, 3);
+
+    switch (get_highest_layer(default_layer_state)) {
+        case _COLEMAK:
+            oled_write(" CLMK ", false);
+            break;
+        case _GAME:
+            oled_write(" GAME ", false);
+            break;
+        case _QWERTY:
+            oled_write("QWERT", false);
+            break;
+        default:
+            oled_write("UNDEF", false);
```

Comment:
Just a heads up, using the "_P" version uses less memory on the system, something of which atmega32u4's have very little of.

```suggestion
            oled_write_P(PSTR(" CLMK "), false);
            break;
        case _GAME:
            oled_write_P(PSTR(" GAME "), false);
            break;
        case _QWERTY:
            oled_write_P(PSTR("QWERT"), false);
            break;
        default:
            oled_write_P(PSTR("UNDEF"), false);
```